internal/host/host.go

```diff
@@ -1275,3 +1278,36 @@ func (m *Manager) GetHostByKubeKey(key types.NamespacedName) (*common.Host, erro
 func (m *Manager) UnRegisterHost(ctx context.Context, hostID, infraEnvID string) error {
 	return common.DeleteHostFromDB(m.db, hostID, infraEnvID)
 }
+
+func (m *Manager) GetReadyAndApprovedHostsSummary(clusterID strfmt.UUID) (registered, approved int, err error) {
+	var hostCounts []struct {
+		Count    int
+		Approved bool
+	}
+	err = m.db.Table("hosts").Select("count(*) as count, approved").Group("approved").
+		Where("cluster_id = ? and status = ? and deleted_at is null", clusterID.String(), models.HostStatusKnown).Scan(&hostCounts).Error
+	if err != nil {
+		m.log.WithError(err).Errorf("Failed to get counts")
+		return
+	}
+	for _, h := range hostCounts {
+		registered += h.Count
+		if h.Approved {
+			approved += h.Count
+		}
+	}
+	return
+}
+
+func (m *Manager) HostWithCollectedLogsExists(clusterId strfmt.UUID) (bool, error) {
+	var hostExists []int
+	if err := m.db.Raw(`select 1 as host_exists where exists (select id from hosts where cluster_id = ? and 
+		logs_collected_at <> ? and deleted_at is null)`, clusterId.String(), time.Time{}).Pluck("host_exists", &hostExists).Error; err != nil {
```

Comment:
Can you make a separate variable for the static string part of this query and format it like an actual SQL query to make it easier to read?

Will something like this work? Or will the whitespace mess things up?
Also could this query be simplified to remove the subquery?
```go
query := `SELECT 1 as host_exists
          FROM hosts
          WHERE
            cluster_id = ? AND
            logs_collected_at <> ? AND
            deleted_at IS NULL`
if err := m.db.Raw(query, clusterId.String(), time.Time{}).Pluck("host_exists", &hostExists).Error; err != nil {
```